### PR TITLE
refactor: Use optimus config struct instead of interface

### PR DESCRIPTION
**Pull request by deryrahman**
- [x] remove config/interface.go
- [x] remove config.Optimus getters

### CODE REVIEW

**Comment by kushsharma** [comment]
Hi @deryrahman, although this looks good, can you please help me understand the reasoning behind why we are doing this?

**Comment by deryrahman** [comment]
@kushsharma As of now, since we only have 1 config implementation (which is config.Optimus) we don't need the interface. We also saw that the config is only data structure and have no behavior. And all the fields of conf.Optimus struct is exported (public) so we don't see the urgency of having getter. But if there's a strong reason to use getter, then we can discuss

File: cmd/server/server.go [hunk 1]
```diff
@@ -252,18 +252,18 @@ func jobSpecAssetDump() func(jobSpec models.JobSpec, scheduledAt time.Time) (mod
 	}
 }
 
-func checkRequiredConfigs(conf config.Provider) error {
+func checkRequiredConfigs(conf config.Optimus) error {
```
Comment (sbchaos) [inline cmd/server/server.go:255]: Pass only the `config.ServerConfig` to this method.

File: cmd/secret.go [hunk 2]
```diff
@@ -249,14 +249,14 @@ func updateSecret(l log.Logger, conf config.Provider, req *pb.UpdateSecretReques
 	return nil
 }
 
-func listSecret(l log.Logger, conf config.Provider, req *pb.ListSecretsRequest) (err error) {
+func listSecret(l log.Logger, conf config.Optimus, req *pb.ListSecretsRequest) (err error) {
```
Comment (sbchaos) [inline cmd/secret.go:252]: Pass only the `conf.Host` to this method.

File: cmd/secret.go [hunk 3]
```diff
@@ -215,14 +215,14 @@ func registerSecret(l log.Logger, conf config.Provider, req *pb.RegisterSecretRe
 	return nil
 }
 
-func updateSecret(l log.Logger, conf config.Provider, req *pb.UpdateSecretRequest) (err error) {
+func updateSecret(l log.Logger, conf config.Optimus, req *pb.UpdateSecretRequest) (err error) {
```
Comment (sbchaos) [inline cmd/secret.go:218]: Pass only the conf.Host to this method.

File: cmd/secret.go [hunk 4]
```diff
@@ -181,14 +181,14 @@ func validateProperlyEncoded(secretValue string) error {
 	return nil
 }
 
-func registerSecret(l log.Logger, conf config.Provider, req *pb.RegisterSecretRequest) (err error) {
+func registerSecret(l log.Logger, conf config.Optimus, req *pb.RegisterSecretRequest) (err error) {
```
Comment (sbchaos) [inline cmd/secret.go:184]: Pass only the `conf.Host` to this method.

File: cmd/replay_create.go [hunk 5]
```diff
@@ -213,14 +213,14 @@ func printExecutionTree(instance *pb.ReplayExecutionTreeNode, tree treeprint.Tre
 }
 
 func runReplayRequest(l log.Logger, projectName, namespace, jobName, startDate, endDate string, forceRun bool,
-	allowedDownstreamNamespaces []string, conf config.Provider) (string, error) {
```
Comment (sbchaos) [inline cmd/replay_create.go:216]: Pass only the `conf.Host` to this method.

File: cmd/replay_create.go [hunk 6]
```diff
@@ -108,14 +108,14 @@ Date ranges are inclusive.
 }
 
 func printReplayExecutionTree(l log.Logger, projectName, namespace, jobName, startDate, endDate string,
-	allowedDownstreamNamespaces []string, conf config.Provider) (err error) {
+	allowedDownstreamNamespaces []string, conf config.Optimus) (err error) {
```
Comment (sbchaos) [inline cmd/replay_create.go:111]: Pass only the `conf.Host` to this method.

File: cmd/backup_create.go [hunk 7]
```diff
@@ -199,14 +199,14 @@ func runBackupDryRunRequest(l log.Logger, conf config.Provider, backupRequest *p
 	return nil
 }
 
-func runBackupRequest(l log.Logger, conf config.Provider, backupRequest *pb.CreateBackupRequest) (err error) {
+func runBackupRequest(l log.Logger, conf config.Optimus, backupRequest *pb.CreateBackupRequest) (err error) {
```
Comment (sbchaos) [inline cmd/backup_create.go:202]: Pass only the `conf.Host` to this method.

File: cmd/backup_create.go [hunk 8]
```diff
@@ -166,14 +166,14 @@ func extractDescription(description string) (string, error) {
 	return description, nil
 }
 
-func runBackupDryRunRequest(l log.Logger, conf config.Provider, backupRequest *pb.BackupDryRunRequest, backupDownstream bool) (err error) {
+func runBackupDryRunRequest(l log.Logger, conf config.Optimus, backupRequest *pb.BackupDryRunRequest, backupDownstream bool) (err error) {
```
Comment (sbchaos) [inline cmd/backup_create.go:169]: Pass only the `conf.Host` to this method.

**Comment by kushsharma** [comment]
> Hi @kushsharma, As of now, since we only have 1 config implementation (which is config.Optimus) we don't need the interface. We also saw that the config is only data structure and have no behavior. And all the fields of conf.Optimus struct is exported (public) so we don't see the urgency of having getter. But if there's a strong reason to use getter, then we can discuss. cmiiw @sbchaos @sravankorumilli

Make sense, unless we have more than one implementation it is not needed as of now. The idea of having the interface was if we needed to inject a provider like etcd store or [gcloud runtimevars](https://gocloud.dev/howto/runtimevar/) for runtime changes but I agree that is unlikely in near future.

**Review by sbchaos — APPROVED** [review]

**Pull request merged**